clientdb/order.go

```diff
@@ -689,6 +698,29 @@ func deserializeOrderTlvData(r io.Reader, o order.Order) error {
 		o.Details().ChannelType = order.ChannelType(channelType)
 	}
 
+	if allowedNodeIDsBytes, ok := parsedTypes[allowedNodeIDsType]; ok {
+		numberOfItems := len(allowedNodeIDsBytes) / 33
+		o.Details().AllowedNodeIDs = make([][33]byte, numberOfItems)
+		if err := binary.Read(
```

Comment:
This falls back to some reflection magic. Since we know exactly that we're serializing and de-serializing a slice of arrays, should we add two helper functions that then do the explicit 33 byte calculation? Then we can just use `copy()` and don't need `binary.Read()` or a buffer for the reverse.

Reply:
Do you mean flattering the `[][33]byte` to a `[]byte` and then use the default tlv types serialize/deserialize? then add a function to transform from `[]byte` to `[][33]byte`?

Reply:
Yes, exactly. Feels more robust and performant than relying on some reflection code of `binary.Read()`.